TM1py/Services/SubsetService.py

```diff
@@ -95,6 +95,28 @@ def update(self, subset: Subset, private: bool = False, **kwargs) -> Response:
             "/api/v1/Dimensions('{}')/Hierarchies('{}')/{}('{}')",
             subset.dimension_name, subset.hierarchy_name, subsets, subset.name)
         return self._rest.PATCH(url=url, data=subset.body, **kwargs)
+    
+    def make_static(self, subset_name: str, dimension_name: str, hierarchy_name: str = None, 
+                    private: bool = False) -> Response:
+        """ convert a dynamic subset into static subset on the TM1 Server
+        :param subset_name: String, name of the subset
+        :param dimension_name: String, name of the dimension 
+        :param hierarchy_name: String, name of the hierarchy
+        :param private: Boolean
+        :return: response
+        """
+        import json
+        from collections import OrderedDict
+        hierarchy_name = hierarchy_name if hierarchy_name else dimension_name
+        payload = OrderedDict()
+        payload['Name'] = subset_name
+        payload['MakePrivate'] = True if private else False
+        payload['MakeStatic'] = True
+        subsets = "PrivateSubsets" if private else "Subsets"
+        request = "/api/v1/Dimensions('{}')/Hierarchies('{}')/{}('{}')/tm1.SaveAs".format(
```

Comment:
Please use the `format_url` method here. It takes care of potential single quotes in object names.

Reply:
oh yeah, my local version wasn't up to date. Just updated it!